specs/enhance_gluster_cluster_sync_logic.adoc

```diff
@@ -0,0 +1,492 @@
+of, using it for deciding the overall cluster status and volume utilization is
+not appropriate. Rather we need to write the aggregations logic to decide the
+cluster state based on peers status and volume status of the gluster trusted
+storage pool. Also the overall volume utilization should be calculated based on
+aggregations done for the participating bricks of the volumes.
+
+
+== Use Cases
+
+This addresses the synchronization of gluster cluster details like
+
+* Volume snapshot details
+* Add volume re-balance estimation time details
+* Volume geo-rep details
+* Client connections per volume
+* Identification of arbiter volumes in cluster (if any)
+* Instead of `gstatus` use aggregations across bricks utilization to calculate
+volume utilization
+* Instead of `gstatus` use peer status and volume status to aggregate and
+calculate the cluster status
+
+
+== Proposed change
+
+* Use `get-state detail` and `get-state volumeoptions` commands instead of plain
+`get-state` as first provides the detailed (e.g. client connections etc.)
+information for the volume and second is option to list all the volume options
+with default values
+
+* Use `get-state` output to figure out the volume snapshot details and update in
+central store. The snapshot details available as part of `get-state` output are
+documented under references section in sample. These details should be used to
+create instances of `Snapshot` class and persist for a volume.
+
+* Calculate the estimation time for re-balance job. This should be largest
+value among the `time_left` values reported by get-state from different nodes of
+the cluster
+
+* Use `get-state` output for geo-rep session details like no. of geo-rep
+sessions and active geo-rep sessions and update per volume in central store
+
+* Figure out the client connections per brick from `get-state` output and
+aggregate for volume and update in central store. The no of client connections
+for a volume merely a sum of all the participating brick's client connections.
+Note: This logic for calculating client connections for a volume is trivial but
+its as per the guidelines provided from Gluster team.
+
+* Identify a arbiter volume and mark accordingly in central store, based on
+field `volume{n}.arbiter_count` in `get-state` output. For a volume if value of
+`volume{n}.arbiter_count` is non-zero, the volume should be marked as arbiter
+type
+
+* Remove the uses of `gstatus` and use aggregations across bricks of the volume
+to decide the volume utilization data.
+
+The logic here goes like
+
+** First calculate the `raw_capacity` and `raw_used` values for volume and
+cluster merely by adding the `raw_capacity` and `raw_used` for bricks of the
+volume. These details are synced and available with central store for the
+individual bricks
+
+** Now the pseudo code to decide the final values of volume and cluster usable
+and used capacity goes as below
+
+```
+if no of up bricks == no of total bricks:
```

Comment:
@atinmu below is the pseudo code for calculation of volume and cluster utilization. The logic is simple copy from `gstatus` at the moment. Need your inputs to figure if its correct or some changes would be required in this logic.
May be if required you can add others from Gluster team to review the logic as well.